diff -upr a/src/libfakechroot.h b/src/libfakechroot.h
--- a/src/libfakechroot.h	2023-12-27 14:34:49.611127202 +0100
+++ b/src/libfakechroot.h	2023-12-27 14:35:52.055127326 +0100
@@ -29,6 +29,22 @@
 #include "rel2abs.h"
 #include "rel2absat.h"

+#ifndef _STAT_VER
+ #if defined (__aarch64__)
+  #define _STAT_VER 0
+ #elif defined (__powerpc__) && __WORDSIZE == 64
+  #define _STAT_VER 1
+ #elif defined (__riscv) && __riscv_xlen==64
+  #define _STAT_VER 0
+ #elif defined (__s390x__)
+  #define _STAT_VER 1
+ #elif defined (__x86_64__)
+  #define _STAT_VER 1
+ #else
+  #define _STAT_VER 3
+ #endif
+#endif
+

 #define debug fakechroot_debug

